src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,66 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_CAPACITY = 16;
+    static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private MyEntry<K, V>[] values = new MyEntry[DEFAULT_CAPACITY];
 
     @Override
     public void put(K key, V value) {
-
+        boolean insert = true;
+        for (int i = 0; i < size; i++) {
+            if (values[i].key != null) {
+                if (values[i].key.equals(key)) {
+                    values[i].value = value;
+                    insert = false;
+                }
+            } else if (values[i].key == null && key == null) {
+                values[i].value = value;
+                insert = false;
+            }
+        }
+        if (insert) {
+            ensureCap();
+            values[size++] = new MyEntry<>(key, value);
+        }
     }
 
     @Override
     public V getValue(K key) {
+        for (int i = 0; i < size; i++) {
+            if (values[i] != null && values[i].key != null) {
+                if (values[i].key.equals(key)) {
+                    return values[i].value;
+                }
+            } else if (key == null && values[i].key == null) {
+                return values[i].value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void ensureCap() {
+        if (size == values.length * LOAD_FACTOR) {
+            int newSize = values.length * 2;
+            values = Arrays.copyOf(values, newSize);
+        }
```

Comment:
when making a bigger table of entities `hash % 16 != hash % 32`
so entries should be placed in different bucket
![image](https://user-images.githubusercontent.com/81781479/196037036-5558dd29-8853-4be1-b7e8-5d53c4e174d8.png)